### PR TITLE
Models/logistic regression mord orf

### PR DESCRIPTION
**Pull request by tunczyk101**
Models:

- CORAL & CORN notebook
- Mord: LogisticAT & LogisticIT
- ORF
- Statsmodels

Other Tasks:

- Add scenarios
- Filter the monsters' dataset by name (ensure only one monster with a given name)
- Add spell DC (Difficulty Class)
- Reformat the repo structure
- Add a file with society scenarios (without dates) to include them in the future
- Update poetry configuration (because of ORF)
- Save datasets to CSV

### CODE REVIEW

File: dataset/creating_dataset.py [hunk 1]
```diff
@@ -286,6 +288,18 @@ def get_aoo(items_list: list[dict]):
     return len(reactions)
 
 
+def get_dc_bonus(items_list: list[dict]):
```
Comment (j-adamczyk) [inline dataset/creating_dataset.py:291]: Return type is `tuple[int, int]`, right?

File: dataset/creating_dataset.py [hunk 2]
```diff
@@ -320,6 +334,16 @@ def sort_preprocessed_data(bestiary: pd.DataFrame) -> pd.DataFrame:
     return bestiary
 
 
+def remove_older_monsters_duplicates(df: pd.DataFrame):
```
Comment (j-adamczyk) [inline dataset/creating_dataset.py:337]: Return type is `pd.DataFrame`

File: training/train_and_evaluate_models.py [hunk 3]
```diff
@@ -76,20 +86,17 @@ def get_model_results(
     :param thresholds: List of threshold values to consider for rounding=
     :return: Two lists containing evaluation metrics for different rounding strategies
     """
-    if type(model) != OrderedResultsWrapper:
+    if type(model) == OrderedResultsWrapper:
```
Comment (j-adamczyk) [inline training/train_and_evaluate_models.py:89]: Use `isinstance()` instead of `type() == ...`

File: training/train_and_evaluate_models.py [hunk 4]
```diff
@@ -13,6 +15,10 @@
 )
 
 
+def root_mean_squared_error(y_true, y_pred):
+    return mean_squared_error(y_true, y_pred, squared=False)
```
Comment (j-adamczyk) [inline training/train_and_evaluate_models.py:19]: scikit-learn in new versions has this function, update `pyproject.toml`

File: training/create_model.py [hunk 5]
```diff
@@ -141,6 +145,48 @@ def create_model(classifier_name: str):
...
+            }
+            model = GridSearchCV(
+                estimator=rf,
+                param_grid=hyper_params,
+                scoring=make_scorer(orf_mean_absolute_error, greater_is_better=False),
+                return_train_score=True,
+                n_jobs=-1,
+            )
+        case "logisticAT":
+            hyper_params = [{"alpha": np.linspace(0.0, 1e-3, 100)}]
```
Comment (j-adamczyk) [inline training/create_model.py:165]: Those are very small values, i.e. really strong regularization. Are you sure?

File: training/train_and_evaluate_models.py (hunk 4 above)
Comment (tunczyk101) [reply training/train_and_evaluate_models.py:19]: Impossible -> ORF

Comment (j-adamczyk) [reply training/train_and_evaluate_models.py:19]: ok

**Pull request merged**